hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/HoodieParquetRealtimeInputFormat.java

```diff
@@ -115,10 +115,8 @@ void addProjectionToJobConf(final RealtimeSplit realtimeSplit, final JobConf job
           List<String> fieldsToAdd = new ArrayList<>();
           if (!realtimeSplit.getDeltaLogPaths().isEmpty()) {
             HoodieRealtimeInputFormatUtils.addVirtualKeysProjection(jobConf, realtimeSplit.getVirtualKeyInfo());
-            String preCombineKey = tableConfig.getPreCombineField();
-            if (!StringUtils.isNullOrEmpty(preCombineKey)) {
-              fieldsToAdd.add(preCombineKey);
-            }
+            List<String> preCombineKeys = tableConfig.getPreCombineFields();
+            preCombineKeys.stream().filter(StringUtils::nonEmpty).forEach(fieldsToAdd::add);
```

Comment:
The table config should already be filtering out empty strings, right? I think we can remove this check here

Reply:
Fixed it